src/uct/cuda/cuda_ipc/cuda_ipc_ep.c

```diff
@@ -22,19 +22,30 @@
 #define UCT_CUDA_IPC_PUT 0
 #define UCT_CUDA_IPC_GET 1
 
+
 static UCS_CLASS_INIT_FUNC(uct_cuda_ipc_ep_t, const uct_ep_params_t *params)
 {
     uct_cuda_ipc_iface_t *iface = ucs_derived_of(params->iface,
                                                  uct_cuda_ipc_iface_t);
+    const uct_cuda_ipc_iface_addr_t *iface_addr;
 
     UCT_EP_PARAMS_CHECK_DEV_IFACE_ADDRS(params);
     UCS_CLASS_CALL_SUPER_INIT(uct_base_ep_t, &iface->super);
 
+    iface_addr = (const uct_cuda_ipc_iface_addr_t*)params->iface_addr;
+
+    self->keepalive = uct_ep_keepalive_create(iface_addr->pid,
```

Comment:
Is `uct_cuda_ipc_iface_get_address` guaranteed to be called before this point? What I see in UCP is worker_iface_open calling uct_iface_open and there isn't a call for uct_iface_get_address call before that. Isn't it better to just populate pid and starttime at iface_open call and make `uct_cuda_ipc_iface_get_address` a no-op?

Reply:
> Is uct_cuda_ipc_iface_get_address guaranteed to be called before this point?

yes, it's guaranteed. this iface_address received from a peer (via out of band communication or WIREUP_MSG phase in UCP).
so, it means that all fields should be initialized by our peer.

Reply:
> device

This should be handled by https://github.com/openucx/ucx/pull/5737 here https://github.com/openucx/ucx/pull/5737/files#diff-8b80dd5e092bb2439236b2f818c36fa956a3005025f6029c1ea9ce449d1ddf25R97